### PR TITLE
[serve] Raise error in `test_local_store_recovery`

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

`test_local_store_recovery` in `test_standalone.py` is [flaky](https://flakey-tests.ray.io/#) on Windows:

Screen Shot 2022-03-23 at 5 52 45 PM

It repeatedly fails on `assert check("hello")`, which suppresses errors, making it tough to debug. This change allows `check()` to raise an error for more actionable outputs.

## Related issue number

N/A

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - Modifies `test_local_store_recovery` in `test_standalone.py`.

### CODE REVIEW

File: python/ray/serve/tests/test_standalone.py [hunk 1]
```diff
@@ -550,12 +550,14 @@ def hello(_):
     def world(_):
         return "world"
 
-    def check(name):
+    def check(name, raise_error=False):
```
Comment (edoakes) [inline python/ray/serve/tests/test_standalone.py:553]: why do we need the flag? why not just always raise? don't see it used anywhere else

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/tests/test_standalone.py (hunk 1 above)
Comment (shrekris-anyscale) [reply python/ray/serve/tests/test_standalone.py:553]: `check()` is used in `wait_for_condition()` call near the end of the function. If it always raises, `wait_for_condition()` will fail if `check()` fails on the first try.

**Comment by edoakes** [comment]
@shrekris-anyscale can you please merge master and push again?

**Comment by edoakes** [comment]
lint was failing but I think it's fixed in master

**Pull request merged**